//
// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package scql.engine;

enum DataSourceKind {
  UNKNOWN = 0;
  MYSQL = 1;
  SQLITE = 2;
  POSTGRESQL = 3;
  CSVDB = 4;
  ARROWSQL = 5;
  GRPC = 6;
}

message DataSource {
  // datasource uuid
  string id = 1;
  // human-friendly datasource name
  string name = 2;
  DataSourceKind kind = 3;
  // concrete data source connection string
  // It is comprehend to related data source adaptor.
  string connection_str = 4;
};